extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -948,7 +954,33 @@ private void setupUnixDomainSocketHttpServer(HttpServer httpServer, HttpServerOp
         }
 
         private void setupTcpHttpServer(HttpServer httpServer, HttpServerOptions options, boolean https,
-                Promise<Void> startFuture, AtomicInteger remainingCount) {
+                Promise<Void> startFuture, AtomicInteger remainingCount, AtomicInteger currentConnectionCount) {
+            if (quarkusConfig.limits.maxConnections.isPresent() && quarkusConfig.limits.maxConnections.getAsInt() > 0) {
+                final int maxConnections = quarkusConfig.limits.maxConnections.getAsInt();
+                httpServer.connectionHandler(new Handler<HttpConnection>() {
+
+                    @Override
+                    public void handle(HttpConnection event) {
+                        int current;
+                        do {
+                            current = currentConnectionCount.get();
+                            if (current == maxConnections) {
+                                //just close the connection
+                                LOGGER.debug("Rejecting connection as there are too many active connections");
+                                event.close();
```

Comment:
should we just close or return a 429 response?

Reply:
Technically there may not actually even be a request yet (as you can open the connection before sending the request), so its hard to say. The problem with connection limits is that there is not really any graceful way to handle them.

Reply:
The question is how would the load-balancer in front react to "close". This may even be implementation specific.

@vietj, when you worked on the HTTP proxy for Vert.x, did you look into that?

Reply:
The other thing is that connection limits are probably generally less important that concurrent request limits. If the connections are idle they have very low overhead.